.changeset/honest-moles-divide.md

```diff
@@ -0,0 +1,5 @@
+---
+"@primer/react": patch
+---
+
+Delete TreeView.LinkItem.
```

Comment:
Let's brief migration instructions in the changeset (these get added to the release notes):

````suggestion
Delete `TreeView.LinkItem`. Use `TreeView.Item` and call a navigation function inside the `onSelect` callback instead:

```diff
- <TreeView.LinkItem href="#">...
+ <TreeView.Item onSelect={() => navigate('#')}>...
```
````